utils/stats.py

```diff
@@ -0,0 +1,41 @@
+from functools import wraps
+from time import time
+
+
+class Counter:
```

Comment:
How can we test this?

And apply it to other strats

Reply:
Will add some tests for the Counter class as well. You can apply them wherever needed using the same decorators.